tutorial/practicals/LFRic/building_code/4_psydata/README.md

```diff
@@ -0,0 +1,205 @@
+
+In this example you will use some of the available PSyData
+transformation to instrument the previous LFRic example.
+Initially we will be using kernel extraction as an example,
+but a list of other PSyData applications is provided at the end.
+
+
+## Step 1: Compile all required PSyData libraries
+Change directory to PSyclone's ``lib`` directory, and trigger
+compilation of all PSyData libraries.
+
+    cd $PSYCLONEHOME/lib
+    make all
+
+You need the NetCDF development package installed, the makefiles
+will be using ``nf-config`` to get the appropriate compiler and
+linker flags. Additionally, compiler and compiler flags can be provided
+using the environment variables F90 and F90FLAGS:
+
+    F90=ifort F90flags="-O2 -traceback" make all
+
+By default gfortran will be used.
+
+
+## Step 2: Create a transformation script
+This transformation script will be used as a parameter
+to PSyclone when building the application. The script will apply
+transformation to insert
+code to extract input- and output-data of a kernel. Initially
+we will only transform the kernel in ``time_evolution_alg_mod.x90``.
+Use the simple script ``transform_one.py`` as a template
+to create a new transformation script that applies the
+LFRic-specific kernel extraction transformation. The transformation
+can be imported from ``psyclone.domain.lfric.transformations``
+and is called ``LFRicExtractTrans``.
+
+The ``apply()`` function in this script is called by PSyclone before
+any code is created. Inside ``apply`` create an instance of the
+extraction transformation and apply it to the schedule for
+"invoke_propagate_perturbation". The template file ``transform_one.py``
+contains most of the required calls, you only need to fill in the
+details.
+
+## Step 3: Modify the makefile so that PSyclone invokes the script
+
+There is a set of makefiles provided in this directory, one for
+each of the available PSyData transformation. They all include
+``Makefile.inc`` which defines all of the rules. The PSyData-specific
+makefiles just supply different settings for compiling and linking
+with the PSyData libraries, and they contain the rule to run PSyclone
+on the file ``time_evolution_alg_mod.x90``.
+
+For this part of the exercise add the ``-s`` flag to the ``psyclone``
+invocation in ``Makefile.extract`` and provide the path to your script
+so that PSyclone will invoke it. You have start with ``./``, otherwise
+Python will not find our file (unless you modify ``PYTHONPATH``).
+You can then create your application using:
+
+    make -f Makefile.extract
+
+If you should get an error message that your script is not found,
+it is possible that it contains a syntax error. You can quickly
+test this by using:
+
+    python ./your_transformation_script.py
+
+If the python command does not return anything, your script is at least
+syntactically correct, otherwise Python will print out a hopefully
+useful error message.
+
+
+## Step 4: Look at the produces code for the PSY layer
```

Comment:
s/produces code for the PSY/generated PSy/

Reply:
Done